compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -93,7 +95,9 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
     // TODO fill in
   }
 
-  class PostTyperTransformer extends Transformer {
+  class PostTyperTransformer(@constructorOnly ictx: Context) extends Transformer {
+
+    @threadUnsafe private val compilingScala2StdLib = ictx.settings.Yscala2Stdlib.value(using ictx)
```

Comment:
no `@threadUnsafe` needed here

Reply:
Why not move the `val` outside `PostTyperTransformer`? Then you don't need to pass a context.

Reply:
I did not have access to any context outside of `PostTyperTransformer`. I could have moved it into `newTransformer`. But now there is a better solution.